### PR TITLE
Perform Node 6-16 -> 20 runner upgrade experiment

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Description**
For now, if the telemetry publishing through commandExtension in worker it won't be published without [specific variable](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs#L38) which is needed to block telemetry from third party tasks([related PR](https://github.com/microsoft/azure-pipelines-agent/pull/4522)). The variable sets in [TaskRunner](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/TaskRunner.cs#L102), so, for example, [node telemetry](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/Handlers/NodeHandler.cs#L573) will be published only for in-the-box tasks, because the variable is false.
The PR allows to publish agent worker's telemetry through commandExtension API.

**Related WI**: [AB#2139555](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2139555)

**Testing**:
Tested locally, after the changes handler's telemetry started to publish

### CODE REVIEW

**Review by ivanduplenskikh — APPROVED** [review]

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs [hunk 1]
```diff
@@ -29,13 +35,25 @@ public sealed class PublishTelemetryCommand : IWorkerCommand
         public string Name => "publish";
         public List<string> Aliases => null;
 
+        public bool IsAgentTelemetry;
```
Comment (DergachevE) [inline src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs:38]: Could be done as read-only property with default value of `false`

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs [hunk 2]
```diff
@@ -29,13 +35,25 @@ public sealed class PublishTelemetryCommand : IWorkerCommand
...
+
+        public PublishTelemetryCommand()
```
Comment (DergachevE) [inline src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs:40]: Not needed

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 1 above)
Comment (DmitriiBobreshev) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: Fixed, thanks!

File: src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs (hunk 2 above)
Comment (DmitriiBobreshev) [reply src/Agent.Worker/Telemetry/TelemetryCommandExtension.cs]: Removed

**Review by DergachevE — APPROVED** [review]

**Pull request merged**